### PR TITLE
Adding optional timestamp semantics to SparkHBase

### PR DESCRIPTION
**Pull request by hkothari**

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala [hunk 1]
```diff
@@ -138,7 +143,7 @@ case class HBaseRelation(
         System.arraycopy(x, 0, rBytes, offset, x.length)
         offset += x.length
       }
-      val put = new Put(rBytes)
+      val put = if (timestamp.isDefined) new Put(rBytes, timestamp.get) else new Put(rBytes)
```
Comment (zhzhan) [inline src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala]: val put = timestamp.fold(new Put(rBytes))(new Put(rBytes, _))

File: src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseTableScan.scala [hunk 2]
```diff
@@ -19,6 +19,7 @@ package org.apache.spark.sql.execution.datasources.hbase
 
 import java.util.ArrayList
 
+import com.sun.tools.javac.code.TypeTag
```
Comment (zhzhan) [inline src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseTableScan.scala]: Not used, and failed compilation in jdk1.7

File: src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala (hunk 1 above)
Comment (hkothari) [inline src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseRelation.scala]: Fixed

File: src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseTableScan.scala (hunk 2 above)
Comment (hkothari) [inline src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseTableScan.scala]: Fixed

**Pull request merged**